feat: remove external deps, add project manager and CI config

victorolinasc opened the pull request:
### Brief summary of what the changes does

- removes external dependencies of opinionated configurations (helm, some indicators, etc)
- add [Eldev](https://emacs-eldev.github.io/eldev) project manager to ease package development (build, test, run, lint, etc)
- use project manager to add a CI workflow and test it in supported Emacs versions
- update minimum version to Emacs 28 (two major versions below current latest stable release)
- removed the examples folder as it is outdated (use-package is now part of the core Emacs and has a very good documentation about setting packages)
- removed package declaration from test file (it is not meant to be distributed)

This is a foundational PR so that we can build upon for resolving other issues. As I see current things in the Emacs ecosystem, major modes are meant for purely:

- syntax understanding of a language (either regex based or treesitter based)
- font locking
- indentation
- imenu

Other "IDE" features are usually added beside the major modes. For example, you can have folding, snippets, and etc without the need to add dependencies to the major mode. Nowadays, a major mode paired with a language server gives pretty much everything someone would want from a more traditional "IDE" environment.

If time allows and if my mind can make sense of things, I plan to add a treesitter based mode here too that, with conditional compilation, can be enabled preferably. 

If we want to point users to a "complete environment" for coding Pony under Emacs, maybe we can just indicate it on the README about the language server configuration with Eglot (which is also built-in currently since Emacs 29). 

This would need some breaking documentation somewhere so that people relying on these indicators, helm and so on can still use them with extra configuration outside the major mode.

### Checklist

Please confirm with `x`:

- [x] I own the copyright to the submitted changes and indemnify `ponylang-mode` from any copyright claim that might result from my not being the authorized copyright holder.
- [x] I've read [CONTRIBUTING.md](https://github.com/ponylang/ponylang-mode/blob/main/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback -- this is the default for Eldev
- [X] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [ ] I have confirmed some of these without doing them

SeanTAllen commented on ponylang-mode.el:
```diff
@@ -1,10 +1,10 @@
 ;; URL: https://github.com/ponylang/ponylang-mode
 ;; Keywords: languages programming
-;; Package-Requires: ((emacs "25.1") (dash "2.17.0") (hydra "0.15.0") (hl-todo "3.1.2") (yafolding "0.4.1") (yasnippet "0.14.0") (company-ctags "0.0.4") (rainbow-delimiters "2.1.4") (fill-column-indicator "1.90"))
+;; Package-Requires: ((emacs "28.1"))
```
this says 28.1 and the README 29+. that looks incorrect.

SeanTAllen commented on test/ponylang-mode-test.el:
```diff
@@ -1,29 +1,4 @@
-;;; ponylang-test.el --- Tests for ponylang-mode
-;;
-;; Author: Sean T Allen <sean@monkeysnatchbanana.com>
-;; Version: 0.6.0
-;; URL: https://github.com/seantallen/ponylang-mode
-;; Keywords: programming
-;; Package-Requires: ()
-;;
-;; This file is not part of GNU Emacs.
-;;
-;; Copyright (c) 2015 Austin Bingham
-;; Copyright (c) 2016 Sean T. Allen
-;; Copyright (c) 2020 Damon Kwok
-;; Copyright (c) 2021 The Pony Developers
-;;
-;;; Commentary:
-;;
-;; Description:
-;;
-;; Tests for ponylang-mode
-;;
-;; The tests are built using the standard `ert' module, so you can run
-;; them using a number of techniques. For example:
-;;
-;;     (require 'ponylang-test)
-;;     (ert-run-tests-interactively "ponylang-test")
```
why the removal of all this?

victorolinasc replied:
As a test file, it is not supposed to be shipped with the package and because it has some properties like a real package, Eldev thinks it is another package in the same repository. 

I will re-add the commentary and update it to include eldev instructions it you think it is pertinent but I think we shouldn't include these package attributes, eg: `Author:`, ` Version:`, etc.

victorolinasc commented on .github/workflows/pr.yml:
```diff
@@ -31,3 +31,30 @@ jobs:
+      matrix:
+        emacs_version:
+          # Add more lines like this if you want to test on different Emacs versions.
+          - 28.2
```
Yes it should! And it wilL!

victorolinasc commented on README.md:
```diff
@@ -85,7 +80,7 @@ use something like this:
 ### Using [use-package](https://github.com/jwiegley/use-package)
 
-<!-- If you're using -->
+<!-- If you're using Emacs 29+ or have installed -->
```
After I've re-read everything here I understand the confusion. The thing here is that `use-package`, a tool for managing your Emacs package configuration is built-in the standard Emacs from version 29 and up. So, you don't need to install `use-package` from that version over but you *can* install it in version 28 as an external tool. 

`use-package` became the standard tool by the community and later became part of the core on version 29+.

victorolinasc commented:
I've updated it according to feedback. It is actually version 28+ that we support here. The thing is that from version 29+ `use-package` is built-in into Emacs and, therefore, no need to install it as an external dependency.

I've also updated the screenshot to reflect a pure Emacs session (started with `emacs -q`) and enabling the package only on a Pony example file (it its the hello world from [Jennet](https://github.com/Theodus/jennet) but I can use any other examples if you'd like). The screenshot currently shows the window menu used for many commands. 

Thanks once again for the review! I can update/test anything else you think is necessary here.
